File: modules/core/src/main/scala/symphony/schema/scaladsl/annotations/GQLInputName.scala
package symphony.schema.scaladsl.annotations

import scala.annotation.StaticAnnotation

/**
 * Annotation used to customize the name of an input type.
 * This is usually needed to avoid a name clash when a type is used both as an input and an output.
 */
case class GQLInputName(name: String) extends StaticAnnotation

File: modules/annotations/src/main/java/symphony/annotations/java/GQLInputName.java
package symphony.annotations.java;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Annotation used to customize the name of an input type. This is usually needed to avoid a name
 * clash when a type is used both as an input and an output.
 */
@Documented
@Retention(RetentionPolicy.CLASS)
@Target({ElementType.TYPE})
public @interface GQLInputName {
  String value() default "";
}
